types/project.go

```diff
@@ -572,7 +618,15 @@ func (p Project) ResolveServicesEnvironment(discardEnvFiles bool) error {
 		if discardEnvFiles {
 			service.EnvFiles = nil
 		}
-		p.Services[i] = service
+		newProject.Services[i] = service
 	}
-	return nil
+	return newProject, nil
+}
+
+func (p *Project) projectDeepCopy() (*Project, error) {
```

Comment:
why not just `deepCopy`?

Reply:
![useless gif](https://media1.giphy.com/media/OM2GUUqiOG7II/giphy.gif?cid=ecf05e470q8y3vn1nbdhycqi0zncenw3rvwh5ene5713s8xm&ep=v1_gifs_search&rid=giphy.gif&ct=g)